web/packages/teleterm/src/ui/AccessRequestCheckout/useAccessRequestCheckout.ts

```diff
@@ -171,12 +198,34 @@ export default function useAccessRequestCheckout() {
     return workspaceAccessRequest.getCollapsed();
   }
 
-  function toggleResource(
+  async function toggleResource(
     kind: ResourceKind,
     resourceId: string,
     resourceName: string
   ) {
-    workspaceAccessRequest.addOrRemoveResource(kind, resourceId, resourceName);
+    if (kind === 'role') {
+      await workspaceAccessRequest.addOrRemoveRole(resourceId);
+      return;
+    }
+
+    if (
+      kind === 'app' ||
+      kind === 'db' ||
+      kind === 'kube_cluster' ||
+      kind === 'node'
+    ) {
+      await workspaceAccessRequest.addOrRemoveResource(
+        toResourceRequest({
+          kind,
+          resourceId,
+          resourceName,
+          clusterUri,
+        })
+      );
+      return;
+    }
```

Comment:
Here we're dealing with essentially two groups of enum values. I wonder if it wouldn't be better to drop the second `if`: `workspaceAccessRequest.addOrRemoveResource` already enforces only certain `kind`s through its types.

So if a new kind of resource gets added, it'll be one less place to adjust without sacrificing type safety.

Reply:
Actually, it turned out that the type `ResourceKind` from `teleterm/useNewRequest` was too wide, it contained kinds `user_group` and `windows_desktop` which we don't support at all.
After I corrected this, I was able to get rid of this `if`, since we narrowed down the type to `'app' | 'db' | 'kube_cluster' | 'node' anyway.